Add config parameters for RabbitMQ, Spring email and Swift

fozboz opened the pull request:
This PR includes commits that introduce new columns into the db models, allowing for extra configuration parameters that are required if one wants to use RabbitMQ as an SQS replacement, Spring email as an SES/SNS replacement, or Swift object storage as a replacement for duplication policy/audit log storage.

It also includes a commit that updates the main POM to use HTTPS.

Commit:
Add config requirements for RabbitMQ and Spring email

adrwhong committed:
Add Swift parameters to mill database

Commit:
Use HTTPS for Maven central

Also updates organization from DuraSpace to LYRASIS and reference urls
to https

Commit:
Explicitly set column name for s3Type

The automatic parsing of field names that normally translates
myAwesomeField into my_awesome_field does not work for s3Type,
presumably because of the number 3 in the name. So we must explicitly
set it.

bbranan commented:
I performed a rebase on the develop branch to pick up the change that will let the Travis build run.

bbranan commented:
Discussion copied from Slack:

Questions (@bbranan ):
1. How is the s3Type value being used? It would be helpful to know what the possible values are.
2. Why is there a need to add AWS credentials to the DB? We've tried to avoid this where possible. Are these used to contain Swift credentials as well?
3. Most of these columns are set to nullable = false. Clearly there won't be a need for any of these values in my context. How is that being handled?

Response (@fozboz ):
1. This is used to determine if the user has selected Swift or AWS as the location for audit logs (so possible values are "swift" or "aws")
2. This is actually only used for Swift credentials. We decided to use AWS as the label so that we could re-use the constants set in ConfigConstants.java in mill. It might be better to add new constants to avoid confusion.
3. The application sets them to 'N/A' if nothing is specified

Response (@bbranan ):
1. If s3Type is only used for audit logs, what about using something like auditStoreType (which fits with auditQueueType). The added benefit is that you also wouldn't need to set the name in the annotation.
2. Yeah, sounds like the constants approach would be better. I'd like to avoid adding AWS keys in the DB if it's not really necessary.
3. Using `N/A` seems a bit like an unnecessary workaround. Why not just let the fields be nullable = true and add some validation when those fields are really needed

Commit:
Remove Swift props from DB

We are going to read these properties from the .properties file instead.

Made RabbitMQ properties nullable.

Commit:
Make InstanceNotificationTopicARN nullable

In cases where NotifierType is RabbitMQ, this value needs to be set to
Null.

Commit:
Change auditQueueType to queueType

In an effort to reduce the number of "types" that we have.

dbernstein reviewed (approved):
Looks good!
